core/unit_test/TestExecutionEnvironmentNonInitializedOrFinalized.cpp

```diff
@@ -140,4 +140,67 @@ TEST_F(ExecutionEnvironmentNonInitializedOrFinalized_DeathTest, views) {
       matcher2);
 }
 
+TEST_F(ExecutionEnvironmentNonInitializedOrFinalized_DeathTest,
+       c_style_memory_management) {
+  ::testing::FLAGS_gtest_death_test_style = "threadsafe";
+
+  EXPECT_DEATH(
+      {
+        void* ptr = Kokkos::kokkos_malloc(1);
+        Kokkos::kokkos_free(ptr);
+      },
+      "Kokkos allocation \"no-label\" is being allocated before "
+      "Kokkos::initialize was called");
```

Comment:
Suggested improved error message:
```
Kokkos ERROR: attempting to perform C-style memory management via kokkos_malloc() **before** Kokkos::initialize() was called
```
I think that the label is not critical.
Also if we keep the error message quite generic and just name the function being called, the implementation of your internal helper function that checks precondition becomes trivial and just insert the function name.